### DISCUSSION TITLE
Move `internal/encoding` to `encoding` ?

### DISCUSSION THREAD

**Discussion by shykes** [cuelang/cue#448] [General] [answered]
I'm trying to produce text output from a `cue.Instance` similar to that of `cue eval` or `cue def`. This does not seem possible at the moment using the public Go API. The closest equivalent is `cue.Value.Syntax` followed by `cue/format.Node`, but the output is not the same as `cue eval`. Notably, the output cannot always be piped into `cue eval -`.

From reading the source code, I see that `cue eval` and `cue def` use the `cue/internal/encoding` package. At first glance the API seems quite nice, and a perfect fit for me. But it's an internal package... Which makes it a terrible pain to use.

Would it be possible to move `internal/encoding` to `encoding`, as-is? Perhaps with a warning that it might break in the future, similar to `cue/build`. This would make my life much easier...

**Comment by verdverm** [comment]
One thing to note is that `cue eval` is slated for removal. It was trying to be both def and export, but @mpvl didn't end up liking the way it came out and also how it complicates _? (Does top work here as a concept outside of Cue, not that eval complicated the top in Cue?)

This is the last I heard about it anyway

**Comment by mpvl** [comment, marked as answer]
I'm generally fine with that. I wan't sure whether the constants were in the right package and whether this should be only exposed as a higher-level construct in the `load` package. 

We also wanted to change `ParseFile(filename string, src interface{})` signature to something like `Parse(src Source)` to allow something like `Parse(build.Bytes(filename, b))`, for instance.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/448.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
